modules/storage-api/src/main/java/org/apache/ignite/internal/storage/index/IndexRowSerializer.java

```diff
@@ -17,17 +17,17 @@
 
 package org.apache.ignite.internal.storage.index;
 
-import org.apache.ignite.internal.storage.SearchRow;
+import org.apache.ignite.internal.storage.RowId;
 
 /**
  * Temporary API for creating Index rows from a list of column values. All columns must be sorted according to the index columns order,
- * specified by the {@link SortedIndexDescriptor#indexRowColumns()}.
+ * specified by the {@link SortedIndexDescriptor#indexColumns()}.
  */
-public interface IndexRowFactory {
+public interface IndexRowSerializer {
     /**
      * Creates an Index row from a list of column values.
      */
-    IndexRow createIndexRow(Object[] columnValues, SearchRow primaryKey);
+    IndexRow createIndexRow(Object[] columnValues, RowId rowId);
```

Comment:
This approach is inherently ineffective, most of column values could be copied directly from BinaryRow's payload. Serialization to Object and back is redundant

Reply:
This API is temporary and should be removed as soon as we understand how it will be integrated with the partition storage